src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -2292,6 +2292,10 @@ def readCustomCategoryInfo(self):
             plugin_list.append([name, True])
         if plugin_list:
             self.master_category_dict[CATEGORY_CUSTOM] = plugin_list
+        # Adding plugins classified as structure factor to 'CATEGORY_STRUCTURE' list
+        plugin_structure_list = [[name, True] for name, plug in self.custom_models.items() if plug.is_structure_factor]
+        if plugin_structure_list:
+            self.master_category_dict[CATEGORY_STRUCTURE].extend(plugin_structure_list)
```

Comment:
I've been looking into some fixes, but can't seem to recreate this situation when running on my system. When changing a model, combo boxes don't seem to update automatically. To implement the changes I've made, I need to open a new FitPage in the Fit panel. This regenerate all the combo boxes from scratch in the new FitPage, and so each of the plugin models labeled as a structure factor only shows up once. The combo boxes in the FitPages that were open prior to model changes never seem to update.

Do you know where else in the gui `onCustomModelChanged` might be used? Or maybe it should be implemented somewhere and isn't yet? I can also call this method when writing some additional unit tests for this case, but wanted to understand where it is currently being used.

Reply:
Paul meant `onCustomModelChange` which is a slot for the `customModelDirectoryChanged` signal used by both the plugin model editor and Add/Multiply model functionality.

Reply:
Let me be clear that I didn't actually observe any misbehaviour, only code that is suspicious. It may be that all paths leading to it rebuild the categories structure before running, and so it never appears as a bug.